end-user-application-machines/new-agent-vm/Install-Idle-Shutdown.sh

```diff
@@ -0,0 +1,280 @@
+#!/bin/bash
+# Copyright 2017 Teradici Corporation
+
+SERVICE="CAMIdleShutdown.service"
+SERVICE_PATH="/etc/systemd/system/${SERVICE}"
+SERVICE_CONFIG_PATH="${SERVICE_PATH}.d"
+SERVICE_CONFIG="${SERVICE_CONFIG_PATH}/CAMIdleShutdown.conf"
+MONITOR_SCRIPT=/opt/Teradici_CAM_idle_shutdown.sh
+
+function create_monitor_script() {
+    touch ${MONITOR_SCRIPT}
+    chmod +x ${MONITOR_SCRIPT}
+    cat <<'EOF'> ${MONITOR_SCRIPT}
+#!/bin/bash
+# Copyright 2017 Teradici Corporation
+
+
+# Shutdown the machine if it has been idle more than the specified number of minutes
+# Additionally, do not log off if there is an active ssh connection, or
+# there are instances of pcoip-server processes running.
+
+# Global settings
+#
+
+# minimum number of minutes machine must be idle before shutting down
+MIN_IDLE_WAIT_TIME=5
+
+# maximum number of minutes machine must be idle before shutting down
+MAX_IDLE_WAIT_TIME=10000 # about 7 days
```

Comment:
the idle time period and cpu limit and polling interval has a min-max range and we use these to validate and set user provided values to them